MetaMask/metamask-extension
# feat: Enable ppom on other chains

## **Description**

This PR depends on 

1. https://github.com/MetaMask/ppom-validator/pull/90

Once these PRs are merged, and for the Controller PR, a new controller version released, we'll be able to support additional networks on extension and we should enable the feature on them.

The following networks are supported by this change.

Ethereum Mainnet (already done)
BNB chain
Polygon
Arbitrum
Optimism
Avalanche
Linea

**NOTE**: So that we can test this branch, we made some changes in the package.json, and ppom.js once the above 2 dependency PRs are merged, we should revert the changes and especially in the change below, we should be using the deployed version of PPOM Validator.

```
    "@metamask/ppom-validator": "MetaMask/ppom-validator#olu/send-ppom-on-more-networks",
```
## **Related issues**

Fixes: [#1692](https://github.com/MetaMask/MetaMask-planning/issues/1692)
Blocked By: [#21436](https://github.com/MetaMask/metamask-extension/pull/21436), [#90](https://github.com/MetaMask/ppom-validator/pull/90)

## **Manual testing steps**

1. Launch MM
2. Change network to any of the chains above other than Mainnet
3. Open this [modified testdapp](https://seaona.github.io/test-dapp/)
4. Connect MM and click on any of the PPOM buttons
5. You should not see any blockaid banner
6. Checkout this branch, build and restart MM
7. Repeat 2-4
8. You should see blockaid banner when you click on any of the PPOM buttons

## **Screenshots/Recordings**

### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
